hackernews/program.py

```diff
@@ -0,0 +1,59 @@
+"""This project monitors Hacker News for new articles on a specific topic provided via a Slack command
+and posts updates to a Slack channel in real-time."""
+
+import os
+import requests
+import time
+import urllib.parse
+
+from autokitteh.slack import slack_client
+
+
+slack = slack_client("slack_connection")
+SLACK_CHANNEL = os.getenv("SLACK_CHANNEL")
+
+
+def on_slack_command(event):
+    """
+    Workflow's entry-point.
+
+    The function extracts a topic from a Slack command, monitors for new articles,
+    and posts updates to the Slack channel.
+    """
+    topic = event.data.text
+    slack.chat_postMessage(
+        channel=SLACK_CHANNEL, text=f"Waiting for new articles on the topic: {topic}."
+    )
+    current_articles = set()
+    fetch_articles(topic, current_articles)
+
+    # NOTE: For low-traffic topics, it might take a while for new articles to be published,
+    # so users may experience delays in receiving notifications.
+    while True:
+        all_articles = set(current_articles)
+        fetch_articles(topic, all_articles)
+        new_articles = all_articles - current_articles
+
+        if new_articles:
+            for article in new_articles:
+                _, title, url = article
+                s_message = f"Title: {title}, URL: {url if url else 'No URL'}"
+                print(s_message)
+                slack.chat_postMessage(channel=SLACK_CHANNEL, text=s_message)
+            current_articles.update(new_articles)
+
+        time.sleep(120)
+
+
+def fetch_articles(topic, all_articles):
+    encoded_query = urllib.parse.quote(topic)
+    full_url = f"http://hn.algolia.com/api/v1/search_by_date?query={encoded_query}&tags=story&page=0"
```

Comment:
1. This should be a global and "constant" variable, e.g. `API_URL`
2. Query params don't have order in URLs, so you can use `"http://hn.algolia.com/api/v1/search_by_date?tags=story&page=0&query=" and just append the encoded string to it with using f-string formatting
3. What happens in popular topics that have more than 1 page of new results in 2 minutes? And are you sure that the order of the results is stable? In other words, do we need to get more than a single page?